src/Nethermind/Nethermind.Blockchain/HistoryPruning/HistoryPruner.cs

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Blockchain.Receipts;
+using Nethermind.Config;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Logging;
+
+namespace Nethermind.Blockchain.HistoryPruning;
+
+public class HistoryPruner(
+    IBlockTree blockTree,
+    IReceiptStorage receiptStorage,
+    ISpecProvider specProvider,
+    IHistoryConfig historyConfig,
+    ulong secondsPerSlot,
+    ILogManager logManager) : IHistoryPruner
+{
+    private Task? _pruneHistoryTask;
+    private readonly Lock _pruneLock = new();
+    private ulong _lastPrunedTimestamp;
+    private readonly ILogger _logger = logManager.GetClassLogger();
+    private readonly bool _enabled = historyConfig.Enabled;
+    private readonly ulong _epochLength = secondsPerSlot * 32;
+    private readonly ulong _minHistoryRetentionEpochs = 82125;
```

Comment:
This is minimum is valid for mainnet, but might have other values in other chains. i prefer pulling such a value from chainspec